packages/ra-data-graphql-simple/src/buildVariables.js

```diff
@@ -35,6 +35,9 @@ const castType = (value, type) => {
         case 'SCALAR:Boolean':
             return Boolean(value);
 
+        case 'SCALAR:Date':
```

Comment:
Are you aware that people might name this scalar whatever they want? It is not supported out of the box by GraphQL and might be named `DateTime`, `IsoDate` or `Smurf` for what we know. I think this should be documented that this **example** implementation of a graphql data provider expect it to be named `Date` and be an ISO date.

Reply:
Maybe my fix is not good then. We need to serialize date object to ISO to construct the query. Otherwise it get serialized to {} and the info is lost.